man/systemd-measure.xml

```diff
@@ -172,9 +173,26 @@
         <para>If the <option>--public-key=</option> is not specified but <option>--private-key=</option> is
         specified the public key is automatically derived from the private key.</para>
 
+        <para><option>--certificate=</option> can be used to specify an X.509 certificate as an alternative
+        to <option>--public-key=</option> since v256.</para>
+
         <xi:include href="version-info.xml" xpointer="v252"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--private-key=<replaceable>PATH/URI</replaceable></option></term>
+        <term><option>--private-key-source=<replaceable>TYPE[:NAME]</replaceable></option></term>
+        <term><option>--certificate=<replaceable>PATH</replaceable></option></term>
+
+        <listitem><para>As an alternative to <option>--public-key=</option> for the
+        <command>sign</command> command, these switches can be used to sign with an hardware token. The
+        private key option can take a path or a URI that will be passed to the OpenSSL engine or
+        provider, as specified by <option>--private-key-source=</option> as a type:name tuple, such as
+        engine:pkcs11. The specified OpenSSL signing engine or provider will be used to sign.</para>
```

Comment:
hmm, shouldn't there be --certificate-source=?

i.e. that the privkey quite likely might come from tpm2 or pkcs11 is obvious I think. but that the cert also comes from there is not a given, you might also store that locally and not bother pkcs11/tpm2 stuff with it, hence I think we should support configuring the source explicitly for each object?

Reply:
Which engines support a workflow like that? The ones I am aware of do not, you simply pass a certificate file as you always do.

Reply:
systemd-cryptenroll works like that: you point it to a keypair on pkcs11 and it uses both on them, either cert+privkey or pubkey+privkey.

but it's a different usecase i'd claim, the goal was really on physical cards here which typically store a pair (yubikeys do at least), and I wanted them to work as easily as a fido2 key, i.e. you just plug them in, and enroll them, done, without having to know what a certificate even is, as long as there is one on the device.